### ISSUE TITLE
Read-DbaXEFile returns error Could not load file or assembly 'netstandard

### ISSUE THREAD

**Issue by LynnDavis66** [dataplat/dbatools#7789]
### Verified issue does not already exist?

No

### What error did you receive?

PSMessageDetails      : 
Exception             : System.IO.FileNotFoundException: Could not load file or assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or 
                        one of its dependencies. The system cannot find the file specified.
                        File name: 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
                           at XEModuleCore.SqlXEventCmdlet..ctor()
                           at lambda_method(Closure )
                           at System.Management.Automation.CommandProcessor.Init(CmdletInfo cmdletInformation)
                        
                        WRN: Assembly binding logging is turned OFF.
                        To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
                        Note: There is some performance penalty associated with assembly bind failure logging.
                        To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
                        
TargetObject          : 
CategoryInfo          : NotSpecified: (:) [], FileNotFoundException
FullyQualifiedErrorId : System.IO.FileNotFoundException
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Read-DbaXEFile<Process>, C:\Program Files\WindowsPowerShell\Modules\DBAtools\functions\Read-DbaXEFile.ps1: line 106
                        at <ScriptBlock>, X:\SingleServer\xxxxxxxAuditX.ps1: line 70
PipelineIterationInfo : {}

### Steps to Reproduce

 $xe =  Read-DbaXEFile  -Path \\xxxxxx\x$\xxxxxxx\Audit_Logins_0_132760189662120000.xel -EnableException

### Are you running the latest release?

Yes

### Other details or mentions

Error is encountered at line 106 in Read-DbaXEFile. 

If  I run the command out side of DBA tools, it returns the results as expected.

Read-SqlXEvent -FileName 'X:\LoginAudit\xxxxxx_Logins_0_132760189662120000.xel' | Out-GridView

### What PowerShell host was used when producing this error

Windows PowerShell ISE (powershell_ise.exe)

### PowerShell Host Version

Name                           Value                                                                                                                                          
----                           -----                                                                                                                                          
PSVersion                      5.1.14393.4530                                                                                                                                 
PSEdition                      Desktop                                                                                                                                        
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                        
BuildVersion                   10.0.14393.4530                                                                                                                                
CLRVersion                     4.0.30319.42000                                                                                                                                
WSManStackVersion              3.0                                                                                                                                            
PSRemotingProtocolVersion      2.3                                                                                                                                            
SerializationVersion           1.1.0.1

**Comment by wsmelton** [comment]
Can you expand on what you mean by this statement with exact repo steps?

> If I run the command out side of DBA tools, it returns the results as expected.

**Comment by potatoqualitee** [comment]
I think we need to add a test for the path in the parameters, in this case ` \win-sdpasql-001\C$\LoginAudit\` will not exist because of that dollar sign. it needs to be enclosed in single quotes.

**Comment by potatoqualitee** [comment]
Then again, that is an unable to load assembly, not unable to find file 🤔

**Comment by potatoqualitee** [comment]
can you try this in a PowerShell console and not ISE, @LynnDavis66 ? Sometimes the ISE has issues that we cannot overcome and this works for me in the PowerShell console with a similar PowerShell version.

what other modules do you have loaded? (`Get-Module | select name | clip`)

**Comment by LynnDavis66** [comment]
Hi,
I tried running the code in a console only environment and received the same error.  I uninstalled the current version of DBATools and installed version 1.0.124. That fixed the issue and I no longer receive the error. I downloaded the latest version 1.1.19 and installed it and the error is there again. I am having to install manually as our system policy does not allow Internet access from our production severs.
Here is the code I’m testing with. If I use the read-dbaxefile command, it errors out. If I use the native read-sqlxevent it does not error out.
  $Serverq = 'Win-bisssql-d01.msha-inc.com'

$Files = Get-ChildItem -Path \\$Serverq\C$\LoginAudit\<file://$Serverq/C$/LoginAudit/> -Recurse
     Foreach($F in $Files)
     {
     $Name = $F.name
     $Name
    $xe =  Read-DbaXEFile  -Path \\$Serverq\C$\LoginAudit\$Name<file://$Serverq/C$/LoginAudit/$Name> -EnableException  | Out-GridView
  ##    $xe =  Read-sqlxevent  -file \\$Serverq\C$\LoginAudit\$Name<file://$Serverq/C$/LoginAudit/$Name>  | Out-GridView
      }
Audit_Logins_0_132762048999160000.xel
Read-SqlXEvent : Could not load file or assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies.
The system cannot find the file specified.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\functions\Read-DbaXEFile.ps1:106 char:21
+             $enum = Read-SqlXEvent -FileName $currentfile
+                     ~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], FileNotFoundException
    + FullyQualifiedErrorId : System.IO.FileNotFoundException

 $Serverq = 'Win-bisssql-d01.msha-inc.com'

$Files = Get-ChildItem -Path \\$Serverq\C$\LoginAudit\<file://$Serverq/C$/LoginAudit/> -Recurse
     Foreach($F in $Files)
     {
     $Name = $F.name
     $Name
  ##  $xe =  Read-DbaXEFile  -Path \\$Serverq\C$\LoginAudit\$Name<file://$Serverq/C$/LoginAudit/$Name> -EnableException  | Out-GridView
      $xe =  Read-sqlxevent  -file \\$Serverq\C$\LoginAudit\$Name<file://$Serverq/C$/LoginAudit/$Name>  | Out-GridView
      }
Audit_Logins_0_132762048999160000.xel
Audit_Logins_0_132762726717740000.xel
Audit_Logins_0_132762733109580000.xel
Audit_Logins_0_132762926307940000.xel
Audit_Logins_0_132762935561800000.xel

PS C:\Windows\system32>

Thanks,
Lynn




Name
----
dbatools
ISE
Microsoft.PowerShell.Management
Microsoft.PowerShell.Utility

**Comment by wsmelton** [comment]
Since you are using sqlserver module are you doing these test in separate processes? Meaning a new process to test dbatools and a new process to test sqlserver module?

Can you also let us know what .NET framework versions are installed on the server the tests are executed on?

**Comment by LynnDavis66** [comment]
Here is the code and results I have been using to test.
Lynn

Here is the code I’m testing with. If I use the read-dbaxefile command, it errors out. If I use the native read-sqlxevent it does not error out.
  $Serverq = 'Win-bisssql-d01.msha-inc.com'

$Files = Get-ChildItem -Path \\$Serverq\C$\LoginAudit\<file://$Serverq/C$/LoginAudit/> -Recurse
     Foreach($F in $Files)
     {
     $Name = $F.name
     $Name
    $xe =  Read-DbaXEFile  -Path \\$Serverq\C$\LoginAudit\$Name<file://$Serverq/C$/LoginAudit/$Name> -EnableException  | Out-GridView
  ##    $xe =  Read-sqlxevent  -file \\$Serverq\C$\LoginAudit\$Name<file://$Serverq/C$/LoginAudit/$Name>  | Out-GridView
      }
Audit_Logins_0_132762048999160000.xel
Read-SqlXEvent : Could not load file or assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies.
The system cannot find the file specified.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\functions\Read-DbaXEFile.ps1:106 char:21
+             $enum = Read-SqlXEvent -FileName $currentfile
+                     ~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], FileNotFoundException
    + FullyQualifiedErrorId : System.IO.FileNotFoundException

 $Serverq = 'Win-bisssql-d01.msha-inc.com'

$Files = Get-ChildItem -Path \\$Serverq\C$\LoginAudit\<file://$Serverq/C$/LoginAudit/> -Recurse
     Foreach($F in $Files)
     {
     $Name = $F.name
     $Name
  ##  $xe =  Read-DbaXEFile  -Path \\$Serverq\C$\LoginAudit\$Name<file://$Serverq/C$/LoginAudit/$Name> -EnableException  | Out-GridView
      $xe =  Read-sqlxevent  -file \\$Serverq\C$\LoginAudit\$Name<file://$Serverq/C$/LoginAudit/$Name>  | Out-GridView
      }
Audit_Logins_0_132762048999160000.xel
Audit_Logins_0_132762726717740000.xel
Audit_Logins_0_132762733109580000.xel
Audit_Logins_0_132762926307940000.xel
Audit_Logins_0_132762935561800000.xel

PS C:\Windows\system32>

Thanks,
Lynn

**Comment by LynnDavis66** [comment]
No, I’m not using separate processes.
.net is shown below.
Thanks,
Lynn

***@***.***

**Comment by wsmelton** [comment]
I don't see the dot net versions. GitHub truncates emails being sent.

Please perform your test in a process that doesn't have sqlserver module loaded.

**Comment by potatoqualitee** [comment]
perhaps i can set a config to use `sqlserver\Read-SqlXEvent` but that may cause conflicts afterwards since dbatools and sqlserver aren't compat. the old version works bc we hadn't updated the SMO library and could still use the DLL method instead of the command 🤔  It will be interesting to see the .net version.  please run `[System.Runtime.InteropServices.RuntimeInformation]::get_FrameworkDescription()` and let us know the results

**Comment by LynnDavis66** [comment]
.NET Framework 4.7.3820.0
Thanks,
Lynn

**Comment by andreasjordan** [comment]
Are there any new ideas about this?

**Comment by potatoqualitee** [comment]
This one as well. Solution seems complex :/

**Comment by mattcargile** [comment]
I just got the same issue when running on Windows Server 2016 over PSRemoting 

```powershell
Read-SqlXEvent : Could not load file or assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system
cannot find the file specified.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.32\allcommands.ps1:59935 char:21
+             $enum = Read-SqlXEvent -FileName $currentfile
+                     ~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], FileNotFoundException
    + FullyQualifiedErrorId : System.IO.FileNotFoundException
```

I'm piping `Get-ChildItem` output into `Read-DbaXEFile`. Server is on `.NET Framework 4.7.3468.0`.

**Comment by potatoqualitee** [comment]
That's such a good command, it does everything so nicely. But I'm so limited on my DLL knowledge. @mattcargile can you open a new session, load ONLY the latest version of Microsoft's `SqlServer` module, try it there and let me know if it works? If not, then I can file an issue on their GitHub

**Comment by potatoqualitee** [comment]
Sorry to be clear, let me know if `Read-SqlXEvent -FileName $filename` works without dbatools

**Comment by wsmelton** [comment]
You will need to install .NET 4.6 for it likely work.

**Comment by potatoqualitee** [comment]
They are already on  `4.7.3468.0` -- doesn't that encompass 4.6? if not, horror show lol

**Comment by wsmelton** [comment]
No, major versions are not backward compatible in .NET.

**Comment by mattcargile** [comment]
I was also using `Invoke-Command -AsJob` as well as `Start-Job` while over a `PSSession`. I was able to get it to work by calling my code interactively within a `PSSession`. I even switched servers to use `Windows 2012 R2 Standard` which is also on `.NET Framework 4.7.3890.0`. The main difference is running interactively and the Windows Server versions. I was also switch between my local powershell version ( i.e. `pwsh.exe` and `powershell.exe` ). I'm still a bit confused as to what I was doing wrong. I wasn't working with the `SqlServer` module at all.

**Comment by andreasjordan** [comment]
Our second issue about Read-DbaXEFile, and also very stale. Any new ideas?

**Comment by potatoqualitee** [comment]
Yeah, I will ensure that read-dbaxefile works before publishing 2.0. If not, i'll figure out the .net and rewrite it :/

**Comment by potatoqualitee** [comment]
thank you for the tag! came back to do this

**Comment by potatoqualitee** [comment]
this is hard as hell :rage4:

**Comment by potatoqualitee** [comment]
Hello, can you please run the the following command:
```
Install-Module dbatools -AllowPrerelease -Scope CurrentUser
```

then restart powershell, and try again. 

this is prerelease2 of dbatools 2.0, which contains a ton of fixes

**Comment by potatoqualitee** [comment]
yeah, i rewrote this entirely and it'll be fixed in 2.0. I'll add a test to arm64 to see if it works on arm too
